nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -329,22 +329,28 @@ contract StakingEscrow is Issuer {
     * @notice Get the value of locked tokens for active stakers in (getCurrentPeriod() + _periods) period
     * as well as stakers and their locked tokens
     * @param _periods Amount of periods for locked tokens calculation
+    * @param _startIndex Start index for looking in stakers array
+    * @param _maxStakers Max stakers for looking, if set 0 then all will be used
     * @return allLockedTokens Sum of locked tokens for active stakers
     * @return activeStakers Array of stakers and their locked tokens. Stakers addresses stored as uint256
     **/
-    function getAllActiveStakers(uint16 _periods)
+    function getActiveStakers(uint16 _periods, uint256 _startIndex, uint256 _maxStakers)
         external view returns (uint256 allLockedTokens, uint256[2][] memory activeStakers)
     {
         require(_periods > 0);
 
         uint256 stakersLength = stakers.length;
```

Comment:
BTW, just a minor additional comment: this variable should be renamed to `endIndex` or something similar.

Reply:
agree with renaming, it's `endIndex` + 1, which name will be more understandable?

Reply:
In several languages, the end index is assumed to be exclusive, so not included in the array.

Reply:
✔️